app/code/Magento/Sales/Model/Order/Creditmemo/Total/Discount.php

```diff
@@ -26,6 +27,16 @@ public function collect(\Magento\Sales\Model\Order\Creditmemo $creditmemo)
          * basing on how much shipping should be refunded.
          */
         $baseShippingAmount = $this->getBaseShippingAmount($creditmemo);
+
+        /**
+         * If credit memo's shipping amount is set and Order's shipping amount is 0,
+         *      throw exception with differente message
+         */
+        if ( $baseShippingAmount && $order->getBaseShippingAmount() <= 0 ) {
```

Comment:
```suggestion
        if ($baseShippingAmount && $order->getBaseShippingAmount() <= 0) {
```
according to our code style, if statement condition should be without spaces